LiveChat/index.js

```diff
@@ -3,35 +3,391 @@ import { View, Dimensions, StyleSheet, Image } from 'react-native';
 import PropTypes from 'prop-types';
 import ChatBubble from './components/ChatBubble';
 import Chat from './components/Chat';
-GLOBAL.cookie = '';
-import { init } from "./sdk/livechat-visitor-sdk.min";
+import { init } from "@livechat/livechat-visitor-sdk";
+import { AuthWebView } from "@livechat/customer-auth";
+import { init as CustomerSdkInit } from "@livechat/customer-sdk";
+import * as lc3Parsers from './lc3Parsers'
+import * as lc2Parsers from './lc2Parsers'
 
 const chatIcon = require('./../assets/chat.png');
 const { width } = Dimensions.get('window');
 
 export default class LiveChat extends Component {
   constructor(props) {
     super(props);
-    GLOBAL.cookie = '';
     this.defineStyles();
 
     this.state = {
       isChatOn: false,
+      protocol: 'lc2',
+      messages: [],
+      users: {},
+      queued: false,
+      queueData: {},
+      isTyping: false,
+      onlineStatus: false,
+      connectionState: 'not_connected',
       bubble: props.bubble ? props.bubble : (
         <View style={this.styles.bubbleStyle}>
           <Image source={chatIcon} style={this.styles.icon} />
         </View>
       ),
     };
 
-    GLOBAL.visitorSDK = init({
-      license: props.license,
-      group: props.group,
+  }
+
+  componentDidMount() {
+    const visitorSDK = init({
+      license: this.props.license,
+      group: this.props.group,
+      appName: 'ReactNative',
+    });
+
+    this.initVisitorSdk(visitorSDK)
+
+    visitorSDK.on('protocol_upgraded', () => {
```

Comment:
this is some beautiful piece of code, valentine's day is coming - i would consider this to be a perfect gift for some lucky gyal